al/auxeffectslot.cpp

```diff
@@ -734,7 +734,7 @@ START_API_FUNC
     switch(param)
     {
     case AL_EFFECTSLOT_GAIN:
-        if(!(value >= 0.0f && value <= 1.0f))
+        if(value < 0.0f || value > 1.0f)
```

Comment:
It's better as it originally was. If `value` is NaN, the original version will catch it, while the new version will pass it (since `nan < 0` and `nan > 1` are both false, it won't be detected as out of range).

Reply:
This is an automated change. Sounds like std::isnan should be used.

Reply:
It's safer to check if the value's not within the valid range/set of values, than to check if it's in the invalid range/set, since the invalid set is unbound. As evidenced by the bug introduced by this change, if a numeric standard adds extant values (e.g. IEEE-754 adding NaN), extra checks need to be added for each that may occur. `std::isnan` makes it three checks instead of two, and who knows what else can show up in the future.

Reply:
OK removed.